include/ncmath/Color.h

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file Color.h
+ * @copyright Jaremie Romer and McCallister Romer 2025
+ */
+#pragma once
+
+#include "ncmath/Vector.h"
+
+namespace nc
+{
+/** @brief Four component color. */
+struct Color : public Vector4
+{
+    using Vector4::Vector4;
+
+    constexpr explicit Color(const Vector4& in)
+        : Vector4{in}
+    {
+    }
+
+    constexpr explicit Color(uint8_t r, uint8_t g, uint8_t b, uint8_t a)
+        : Vector4{(float)r / 255.0f, (float)g / 255.0f, (float)b / 255.0f, (float)a / 255.0f}
+    {
+    }
+
+    static constexpr auto Red()   -> Color { return Color{1.0f, 0.0, 0.0f, 1.0f}; }
+    static constexpr auto Green() -> Color { return Color{0.0f, 1.0, 0.0f, 1.0f}; }
+    static constexpr auto Blue()  -> Color { return Color{0.0f, 0.0, 1.0f, 1.0f}; }
+    static constexpr auto White() -> Color { return Color{1.0f, 1.0, 1.0f, 1.0f}; }
+    static constexpr auto Black() -> Color { return Color{0.0f, 0.0, 0.0f, 1.0f}; }
+    static constexpr auto Clear() -> Color { return Color{0.0f, 0.0, 0.0f, 0.0f}; }
```

Comment:
I don't like "Clear" here very much - because of it's use in the engine as 'zero out'. Even though that's effectively what you're getting back - it's not a verb it's a noun. How about 'Transparent'? 'Transparent' seems more conventional IMO.